### PR TITLE
ENT-472: Add gateway_host stage variable for api deployments

### PR DESCRIPTION
**Pull request by saleem-latif**
Hi @efagin , @mattdrayer , @douglashall 

Please take a look at the PR, I have added a new stage variable named `gateway_host` that will contain the api manager's host.

### CODE REVIEW

File: scripts/aws/deploy.py [hunk 1]
```diff
@@ -130,6 +130,8 @@ def update_stage(client, rest_api_id, stage_name, stage_settings):
                         help="Location of catalog IDA for request routing")
     parser.add_argument("--enterprise-host", required=False, default='',
                         help="Location of enterprise IDA for request routing")
+    parser.add_argument("--gateway-host", required=True,
```
Comment (efagin) [inline scripts/aws/deploy.py]: Please don't make this new field required. By setting `required=True`, you are introducing a breaking change. This change will immediately halt any deployment pipelines that rely on this script. Please also set the default to something innocuous like "localhost".

Comment (saleem-latif) [inline scripts/aws/deploy.py]: I have made `--gateway-host` optional.

**Comment by douglashall** [comment]
@saleem-latif @efagin @mattdrayer How does this change get deployed to stage (to production)? Is there any documentation that explains this?

**Comment by efagin** [comment]
@douglashall for edX's specific instance, we have a release pipeline to do this that uses the scripts in the `scripts/aws` directory (see the README in that subdirectory for more details on that approach). I'll reach out to you directly with the specifics (how to verify, etc) of our use case.

**Pull request merged**